### ISSUE TITLE
Nullness Checker: passing a `null` to `Objects.hashCode` causes false positive

### ISSUE THREAD

**Issue by negora** [typetools/checker-framework#1499]
Hello:

I'm having a problem with the Nullness Checker. The method `Objects.hashCode` of the JDK 8 accepts `null` as its argument, in which case it returns `0`. However, the Checker Framework says that it requires a non-null value:

```
found   : @Initialized @Nullable String
required: @Initialized @NonNull Object
```

I believe that this method might be annotated wrong, Right?

I'm using:

  * Checker Framework: 2.1.14
  * Maven 3.0.5
  * Java: 1.8.0_141

Thank you.

**Comment by negora** [comment]
I've observed that the same occurs with the `Objects.equals` method.

**Comment by smillst** [comment]
The parameter of `Objects.hashCode` is annotated with `@Nullable`. 
(See https://github.com/typetools/checker-framework/blob/master/checker/jdk/nullness/src/java/util/Objects.java#L90)  Can you please provide a reproducable test case following [the instructions](https://checkerframework.org/manual/#reporting-bugs)?

**Comment by mernst** [comment]
I'm closing this issue because the original poster has not responded with a reproducible test case.
We can re-open it if anyone can reproduce the problem.

**Comment by negora** [comment]
I'm sorry such a big delay, but I've doing some experiments in my spare time to reproduce this error again, and I've been unable :S . I guess that I did something wrong. I'm not sure...

Thank you for your patience and apologizes for the inconvenience.

**Comment by negora** [comment]
Tonight I've found the reason of this error, because it has happened to me again (what a coincidence). It has been my fault. I've forgotten to add the following line in the `pom.xml` file:

```xml
<plugin>
  <groupId>org.apache.maven.plugins</groupId>
  <artifactId>maven-dependency-plugin</artifactId>
  <version>3.0.2</version>
  <executions>
    <execution>
      <goals>
        <goal>properties</goal>
      </goals>  
    </execution>
  </executions>
</plugin>
```

I've realized about this problem when I've found this message in the log:

> warning: You do not seem to be using the distributed annotated JDK.  To fix the
>   problem, supply this argument (first, fill in the "...") when you run javac:
>     -Xbootclasspath/p:.../checker/dist/jdk8.jar

It's very probable that I forgot to add that lines before the original error occurred. I hope that it's of help for other people.

Thank you.
